### PR TITLE
landscaper integration

### PR DESCRIPTION
**Pull request by jschicktanz**
**How to categorize this PR?**

/area delivery
/kind enhancement

**What this PR does / why we need it**:
adds artifacts for Landscaper integration (blueprint, resources definition file, component descriptor build script)

co-authored by @enrico-kaack-comp 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
- adds artifacts for Landscaper integration
```

### CODE REVIEW

**Review by ScheererJ — DISMISSED** [review]
/lgtm

**Review by DockToFuture — APPROVED** [review]
/lgtm

File: .landscaper/blueprint/deploy-executions.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                  concurrentSyncs: {{ .imports.controllerRegistration.concurrentSyncs }}
+                {{- else }}
+                concurrentSyncs: 20
+                healthcheck:
+                  concurrentSyncs: 20
+                {{- end }}
+              {{- if .imports.imageVectorOverwrite }}
+              imageVectorOverwrite: |
+                {{- toYaml .imports.imageVectorOverwrite | nindent 16 }}
+              {{- end }}
```
Comment (rfranzke) [inline .landscaper/blueprint/deploy-executions.yaml:64]: Please use `ControllerDeployment`s instead, see https://github.com/gardener/gardener/pull/3995

File: .landscaper/resources.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    compress: true
+    mediaType: "application/vnd.gardener.landscaper.blueprint.v1+tar+gzip"
+---
+type: helm.io/chart
+name: networking-calico-chart
+relation: local
+input:
+  type: "dir"
+  path: "../charts/gardener-extension-networking-calico"
+  compress: true
```
Comment (schrodit) [inline .landscaper/resources.yaml:17]: I think we need also `preserveDir` for helm to properly work

**Review by schrodit — CHANGES_REQUESTED** [review]

File: .landscaper/resources.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
+  type: landscaper.gardener.cloud/blueprint
+  name: networking-calico-controller-registration
+  relation: local
+  input:
+    type: "dir"
+    path: "./blueprint"
+    compress: true
+    mediaType: "application/vnd.gardener.landscaper.blueprint.v1+tar+gzip"
```
Comment (schrodit) [inline .landscaper/resources.yaml:9]:
```suggestion
    mediaType: "application/vnd.gardener.landscaper.blueprint.layer.v1.tar+gzip"
```

**Review by schrodit — CHANGES_REQUESTED** [review]
please update the media type otherwise lgtm

**Review by rfranzke — APPROVED** [review]

**Pull request merged**